crates/primitives/src/trie/nibbles.rs

```diff
@@ -1,290 +1,48 @@
-    }
-
-    /// Truncate the current nibbles to the given length.
-    pub fn truncate(&mut self, len: usize) {
-        self.hex_data.truncate(len)
+impl From<Vec<u8>> for StoredNibblesSubKey {
+    fn from(inner: Vec<u8>) -> Self {
+        Self(StoredNibbles { inner: inner.into() })
     }
 }
 
-#[cfg(test)]
-mod tests {
-    use super::*;
-    use proptest::prelude::*;
-
-    #[test]
-    fn hashed_regression() {
-        let nibbles = hex::decode("05010406040a040203030f010805020b050c04070003070e0909070f010b0a0805020301070c0a0902040b0f000f0006040a04050f020b090701000a0a040b").unwrap();
-        let nibbles = Nibbles::from(nibbles);
-        let path = nibbles.encode_path_leaf(true);
-        let expected =
-            hex::decode("351464a4233f1852b5c47037e997f1ba852317ca924bf0f064a45f2b9710aa4b")
-                .unwrap();
-        assert_eq!(path, expected);
-    }
-
-    #[test]
-    fn pack_nibbles() {
-        for (input, expected) in [
-            (vec![], vec![]),
-            (vec![0xa], vec![0xa0]),
-            (vec![0xa, 0xb], vec![0xab]),
-            (vec![0xa, 0xb, 0x2], vec![0xab, 0x20]),
-            (vec![0xa, 0xb, 0x2, 0x0], vec![0xab, 0x20]),
-            (vec![0xa, 0xb, 0x2, 0x7], vec![0xab, 0x27]),
-        ] {
-            let nibbles = Nibbles::from(input);
-            let encoded = nibbles.pack();
-            assert_eq!(encoded, expected);
-        }
-    }
-
-    proptest! {
-        #[test]
-        fn pack_unpack_roundtrip(input in any::<Vec<u8>>()) {
-            let nibbles = Nibbles::unpack(&input);
-            let packed = nibbles.pack();
-            prop_assert_eq!(packed, input);
-        }
-
-        #[test]
-        fn encode_path_first_byte(input in any::<Vec<u8>>()) {
-            prop_assume!(!input.is_empty());
-            let input = Nibbles::unpack(input);
-            let input_is_odd = input.len() % 2 == 1;
-
-            let compact_leaf = input.encode_path_leaf(true);
-            let leaf_flag = compact_leaf[0];
-            // Check flag
-            assert_ne!(leaf_flag & 0x20, 0);
-            assert_eq!(input_is_odd, (leaf_flag & 0x10) != 0);
-            if input_is_odd {
-                assert_eq!(leaf_flag & 0x0f, *input.first().unwrap());
-            }
-
-
-            let compact_extension = input.encode_path_leaf(false);
-            let extension_flag = compact_extension[0];
-            // Check first byte
-            assert_eq!(extension_flag & 0x20, 0);
-            assert_eq!(input_is_odd, (extension_flag & 0x10) != 0);
-            if input_is_odd {
-                assert_eq!(extension_flag & 0x0f, *input.first().unwrap());
-            }
-        }
+impl Compact for StoredNibblesSubKey {
+    fn to_compact(self, buf: &mut impl bytes::BufMut) -> usize {
+        assert!(self.inner.len() <= 32);
+        let mut padded = vec![0; 32];
+        padded[..self.inner.len()].copy_from_slice(&self.inner[..]);
+        buf.put_slice(&padded);
```

Comment:
i guess it's not much of a difference, but can we copy `self.inner` to `buf`  directly, and just copy the necessary padded 0 bytes from a fixed array.